src/overlays/actors/ovl_En_Bubble/z_en_bubble.c

```diff
@@ -32,73 +31,399 @@ const ActorInit En_Bubble_InitVars = {
+    this->actor.shape.yOffset = 16.0f;
+    this->graphicRotSpeed = 16.0f;
+    this->graphicEccentricity = 0.08f;
+    this->expansionWidth = dim;
+    this->expansionHeight = dim;
+    a = Rand_ZeroOne();
+    b = Rand_ZeroOne();
+    c = Rand_ZeroOne();
+    this->unk210 = 1.0f;
+    this->unk214 = 1.0f;
+    d = (a * a) + (b * b) + (c * c);
+    this->unk1F8.x = a / d;
+    this->unk1F8.y = b / d;
+    this->unk1F8.z = c / d;
+}
+
+u32 func_8089F59C(EnBubble* this) {
+    ColliderInfo* info = &this->colliderSphere.elements[0].info;
+
+    info->toucher.dmgFlags = 8;
+    info->toucher.effect = 0;
+    info->toucher.damage = 4;
+    info->toucherFlags = TOUCH_ON;
+    this->actor.velocity.y = 0.0f;
+    return 6;
+}
+
+u32 func_8089F5D0(EnBubble* this) {
+    EnBubble_SetDimensions(this, -1.0f);
+    return 12;
+}
+
+void EnBubble_DamagePlayer(EnBubble* this, PlayState* play) {
+    play->damagePlayer(play, -this->colliderSphere.elements[0].info.toucher.damage);
+    func_800B8E1C(play, &this->actor, 6.0f, this->actor.yawTowardsPlayer, 6.0f);
+}
+
+s32 EnBubble_Explosion(EnBubble* this, PlayState* play) {
+    u32 i;
+    Vec3f effectAccel;
+    Vec3f effectVel;
+    Vec3f effectPos;
+
+    effectAccel = sEffectAccel;
+    Math_SmoothStepToF(&this->expansionWidth, 4.0f, 0.1f, 1000.0f, 0.0f);
+    Math_SmoothStepToF(&this->expansionHeight, 4.0f, 0.1f, 1000.0f, 0.0f);
+    Math_SmoothStepToF(&this->graphicRotSpeed, 54.0f, 0.1f, 1000.0f, 0.0f);
+    Math_SmoothStepToF(&this->graphicEccentricity, 0.2f, 0.1f, 1000.0f, 0.0f);
+    this->actor.shape.yOffset = ((this->expansionHeight + 1.0f) * 16.0f);
+
+    if (DECR(this->explosionCountdown) != 0) {
+        return -1;
+    }
+    effectPos.x = this->actor.world.pos.x;
+    effectPos.y = this->actor.world.pos.y + this->actor.shape.yOffset;
+    effectPos.z = this->actor.world.pos.z;
+    for (i = 0; i < 20; i++) {
+        effectVel.x = (Rand_ZeroOne() - 0.5f) * 7.0f;
+        effectVel.y = Rand_ZeroOne() * 7.0f;
+        effectVel.z = (Rand_ZeroOne() - 0.5f) * 7.0f;
+        EffectSsDtBubble_SpawnCustomColor(play, &effectPos, &effectVel, &effectAccel, &sEffectPrimColor,
+                                          &sEffectEnvColor, Rand_S16Offset(0x64, 0x32), 0x19, 0);
+    }
+    Item_DropCollectibleRandom(play, NULL, &this->actor.world.pos, 0x50);
+    this->actor.flags &= ~ACTOR_FLAG_1;
+    return Rand_S16Offset(90, 60);
+}
+
+u32 func_8089F8BC(EnBubble* this) {
+    if (DECR(this->explosionCountdown)) {
+        return -1;
+    }
+    return func_8089F5D0(this);
+}
+
+s32 func_8089F908(EnBubble* this) {
+    this->expansionWidth += 1.0f / 12.0f;
+    this->expansionHeight += 1.0f / 12.0f;
+
+    if (DECR(this->explosionCountdown) != 0) {
+        return false;
+    }
+    return true;
+}
+
+void EnBubble_Vec3fNormalizedReflect(Vec3f* vec1, Vec3f* vec2, Vec3f* ret) {
+    f32 norm;
+
+    func_80179F64(vec1, vec2, ret);
+    norm = sqrtf((ret->x * ret->x) + (ret->y * ret->y) + (ret->z * ret->z));
```

Comment:
```suggestion
    norm = SQXYZ(ret);
```
Unless that macro hard requires a local vec not a pointer... wonder if we could make a second macro

Reply:
That macro does indeed require a local vec,